### PR TITLE
Use more IIFE, constexpr inline, string_view

### PR DESCRIPTION
**Pull request by sethrj**

### CODE REVIEW

File: app/demo-loop/LDemoIO.cc [hunk 1]
```diff
@@ -217,98 +217,84 @@ TransporterInput load_input(LDemoArgs const& args)
...
-    }
-    else if (ends_with(args.physics_filename, ".gdml"))
-    {
-        // Load imported_data directly from Geant4
-        imported_data = GeantImporter(
-            GeantSetup(args.physics_filename, args.geant_options))();
-    }
-    else
-    {
+    ImportData imported = [&args] {
```
Comment (esseivaju) [inline app/demo-loop/LDemoIO.cc]:
```suggestion
    const ImportData imported = [&args] {
```

File: app/demo-loop/LDemoIO.cc [hunk 2]
```diff
@@ -367,8 +349,8 @@ TransporterInput load_input(LDemoArgs const& args)
         TrackInitParams::Input input;
         input.capacity = args.initializer_capacity;
         input.max_events = args.max_events;
-        params.init = std::make_shared<TrackInitParams>(input);
-    }
+        return std::make_shared<TrackInitParams>(input);
```
Comment (esseivaju) [inline app/demo-loop/LDemoIO.cc]:
```suggestion
        return std::make_shared<TrackInitParams>(std::move(input));
```

File: src/accel/SharedParams.cc [hunk 3]
```diff
@@ -259,13 +259,16 @@ void SharedParams::initialize_core(SetupOptions const& options)
...
                       "defined in the celeritas::SetupOptions");
 
-    celeritas::GeantImporter load_geant_data(GeantImporter::get_world_volume());
-    // Convert ImportVolume names to GDML versions if we're exporting
-    GeantImportDataSelection import_opts;
-    import_opts.particles = GeantImportDataSelection::em_basic;
-    import_opts.processes = import_opts.particles;
-    import_opts.unique_volumes = options.geometry_file.empty();
-    auto imported = std::make_shared<ImportData>(load_geant_data(import_opts));
+    auto imported = [&options] {
```
Comment (esseivaju) [inline src/accel/SharedParams.cc]:
```suggestion
    auto const imported = [&options] {
```

File: src/accel/SharedParams.cc [hunk 4]
```diff
@@ -332,8 +327,22 @@ void SharedParams::initialize_core(SetupOptions const& options)
...
+
+    // Construct simulation params
+    params.sim = SimParams::from_import(*imported, params.particle);
+
+    // Construct track initialization params
+    params.init = [&options] {
+        TrackInitParams::Input input;
+        input.capacity = options.initializer_capacity;
+        input.max_events = options.max_num_events;
+        return std::make_shared<TrackInitParams>(input);
```
Comment (esseivaju) [inline src/accel/SharedParams.cc]:
```suggestion
        return std::make_shared<TrackInitParams>(std::move(input));
```

File: src/accel/SharedParams.cc [hunk 5]
```diff
@@ -347,7 +356,7 @@ void SharedParams::initialize_core(SetupOptions const& options)
     }();
 
     // Construct along-step action
-    {
+    auto along_step = [&params, &options, &imported] {
```
Comment (esseivaju) [inline src/accel/SharedParams.cc]: I would move the temporary in the lambda and call the lambda inside `insert`
```suggestion
    params.action_reg->insert([&params, &options, &imported] {
```

File: src/accel/SharedParams.cc [hunk 6]
```diff
@@ -357,31 +366,10 @@ void SharedParams::initialize_core(SetupOptions const& options)
...
-    {
-        TrackInitParams::Input input;
-        input.capacity = options.initializer_capacity;
-        input.max_events = options.max_num_events;
-        params.init = std::make_shared<TrackInitParams>(input);
-    }
+        return options.make_along_step(asfi);
+    }();
+    CELER_VALIDATE(along_step, << "along-step factory returned a null pointer");
+    params.action_reg->insert(std::move(along_step));
```
Comment (esseivaju) [inline src/accel/SharedParams.cc]:
```suggestion
        auto const along_step{options.make_along_step(asfi)};
        CELER_VALIDATE(along_step,
                       << "along-step factory returned a null pointer");
        return along_step;
    }());
```

**Review by esseivaju — APPROVED** [review]

**Comment by sethrj** [comment]
Thanks for the review and suggestions!

**Pull request merged**